sql/scan.go

```diff
@@ -603,12 +612,7 @@ func (n *scanNode) filterRow() bool {
 		return false
 	}
 
-	v, ok := d.(parser.DBool)
-	if !ok {
-		n.err = fmt.Errorf("WHERE clause did not evaluate to a boolean")
-		return false
-	}
-	return bool(v)
+	return bool(d.(parser.DBool))
```

Comment:
What happens if the WHERE clause evaluates to `NULL`? I think you should add a test where the WHERE clause is something like `a = NULL`.